test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLogoutTests.java

```diff
@@ -71,4 +79,37 @@ public void customWithUriVars() {
 		assertThat(request.getRequestURI()).isEqualTo("/uri-logout/val1/val2");
 	}
 
+	/**
+	 * spring-restdocs uses postprocessors to do its trick. It will work only if these are merged together
+	 * with our request builders. (gh-7572)
+	 * @throws Exception
+	 */
+	@Test
+	public void postProcessorsAreMergedDuringMockMvcPerform() throws Exception {
+
+		RequestBuilder requestBuilder = logout();
+
+		RequestBuilder defaultRequestBuilder = MockMvcRequestBuilders.get("/");
```

Comment:
If you assign this to `MockHttpServletRequestBuilder` we don't need to do the `instanceof` or casting to `ConfigurableSmartRequestBuilder` or `Mergeable`